src/base/bittorrent/trackerentry.h

```diff
@@ -47,26 +49,21 @@ namespace BitTorrent
             NotWorking = 4
         };
 
-        TrackerEntry() = default;
-        TrackerEntry(const QString &url);
-        TrackerEntry(const lt::announce_entry &nativeEntry);
-        TrackerEntry(const TrackerEntry &other) = default;
-        TrackerEntry &operator=(const TrackerEntry &other) = default;
-
-        QString url() const;
-        Status status() const;
-
-        int tier() const;
-        void setTier(int value);
-
-        int numSeeds() const;
-        int numLeeches() const;
-        int numDownloaded() const;
+        struct Endpoint
+        {
+            int protocolVersion = 1;
```

Comment:
I deliberately decided to make it as simple as possible for now. If later there is a real need to have it as an enum, then we will change it.

Reply:
And I strongly disagree to create yet another nested structure (like `announce_infohash` in libtorrent). It seems to me that having `protocolVersion` field is sufficient, and also more convenient for processing on the upper levels.